### PR TITLE
Add Validator to check Problem Existence

### LINKED ISSUE #12: ProblemSetIds validation
Currently, users can submit problemSetIds that aren't mongo ObjectIds. This leads to a status 500 since mongo throws an error when parsing problemSetIds.

### PR DESCRIPTION
**Pull request by idodin**
Resolves #12

### CODE REVIEW

**Comment by idodin** [comment]
@ValerianClerc Did we want to dynamically report which Problem Set IDs are problematic?

File: src/validators/problem.ts [hunk 1]
```diff
@@ -40,4 +42,10 @@ export function problemValidator(method: string): ValidationChain[] {
             ];
         }
     }
+}
+
+async function validateProblemSetIds(problemSetIds: Array<string>) {
```
Comment (idodin) [inline src/validators/problem.ts:47]: Not sure if there's a cleaner way of checking this...

File: src/validators/problem.ts [hunk 2]
```diff
@@ -40,4 +42,10 @@ export function problemValidator(method: string): ValidationChain[] {
...
+    const promiseArray = problemSetIds.map(async problemSetId => await problemSetDBInteractions.find(problemSetId));
```
Comment (idodin) [inline src/validators/problem.ts:48]: logic here is to map each problemSetId into a Promise<IProblemSetModel> object, wait for those promises to resolve and then make sure none of the resolved values are null.

File: src/validators/problem.ts [hunk 3]
```diff
@@ -40,4 +42,10 @@ export function problemValidator(method: string): ValidationChain[] {
             ];
         }
     }
+}
+
+async function validateProblemSetIds(problemSetIds: Array<string>) {
+    const promiseArray = problemSetIds.map(async problemSetId => problemSetDBInteractions.find(problemSetId));
```
Comment (ValerianClerc) [inline src/validators/problem.ts:48]: remove `async` keyword, not necessary

**Pull request merged**